cedar-policy-generators/src/schema.rs

```diff
@@ -641,19 +643,24 @@ impl Bindings {
     // Add a `json_schema::Type` and `UnreservedId` binding.
     // Note that this function always succeeds even if the `UnreservedId` already exists.
     // Under that situation, we create a new `UnreservedId` based on the existing `UnreservedId`.
-    fn add_binding(&mut self, binding: (json_schema::Type<ast::InternalName>, UnreservedId)) {
+    fn add_binding(&mut self, binding: (json_schema::Type<ast::InternalName>, CommonTypeId)) {
         let (ty, id) = binding;
+        let id: UnreservedId = id.into();
         // create a new id when the provided id has been used
         let id_smolstr: &str = id.as_ref();
         let new_id = if self.ids.contains(id_smolstr) {
             let mut new_id = id.to_string();
             while self.ids.contains(new_id.as_str()) {
                 new_id.push('_');
+                new_id.push('_');
```

Comment:
I'm looking at this again because of a DRT failure on this target in 3.4 and don't understand why you need to push `_` three times here.

Reply:
IIRC, adding three `_`s makes sure that an id is valid. Counterexamples for adding one time and two times are `_cedar` and `cedar`.

Reply:
so probably not relevant to 3.4 because we reserved names in 4?

Reply:
Correct.